src/core/ext/transport/chttp2/transport/stream_map.cc

```diff
@@ -112,28 +122,28 @@ static void** find(grpc_chttp2_stream_map* map, uint32_t key) {
     }
   }
 
+  GPR_DEBUG_ASSERT(!strict_find);
   return nullptr;
 }
 
 void* grpc_chttp2_stream_map_delete(grpc_chttp2_stream_map* map, uint32_t key) {
-  void** pvalue = find(map, key);
-  void* out = nullptr;
-  if (pvalue != nullptr) {
```

Comment:
is removing this if safe?

Reply:
Yes, we should only be calling delete on things that we know we added otherwise it's an implementation bug that needs to be fixed. There is a GPR_DEBUG_ASSERT(pvalue != nullptr) I added to catch if it is null during development.